osc-server/src/main/java/org/osc/core/broker/service/securitygroup/BindSecurityGroupService.java

```diff
@@ -440,7 +444,7 @@ private void checkVirtualSystemRedundancyInServiceFunctionChains(EntityManager e
         List<SecurityGroup> sgList = SecurityGroupEntityMgr.listSecurityGroupsBySfcIdAndProjectId(em, sfcId,
```

Comment:
i think larkins has a query to list other SG's by id and project id....you can probably create this method the same way
see
https://github.com/opensecuritycontroller/osc-core/pull/517/files#diff-cb0bcdaa1792493a7aceb46f5207ce59R129

Reply:
Yes we will take off any querying redundant with integration test...For now we will have it.